domains/transfer/Transfer.API/API/Transfer/Api/[Features]/AcceptTransferAgreementProposalCommand.cs

```diff
@@ -0,0 +1,183 @@
+        {
+            throw new BusinessException("Only the receiver company can accept this Transfer Agreement Proposal");
+        }
+
+        if (proposal.EndDate is not null && proposal.EndDate < UnixTimestamp.Now())
+        {
+            throw new BusinessException("This proposal has run out");
+        }
+
+        proposal.ReceiverCompanyTin = command.ReceiverOrganizationTin;
+
+        var taRepo = _unitOfWork.TransferAgreementRepo;
+
+        var hasConflict = await taRepo.HasDateOverlap(proposal, cancellationToken);
+        if (hasConflict)
+        {
+            throw new TransferAgreementConflictException();
+        }
+
+
+        var receiverOrganizationId = command.ReceiverOrganizationId;
+        var wallets = await _walletClient.GetWallets(receiverOrganizationId, CancellationToken.None);
+
+        var walletId = wallets.Result.FirstOrDefault()?.Id;
+        if (walletId == null)
+        {
+            var createWalletResponse = await _walletClient.CreateWallet(receiverOrganizationId, CancellationToken.None);
+
+            if (createWalletResponse == null)
+            {
+                throw new Exception("Failed to create wallet.");
+            }
+
+            walletId = createWalletResponse.WalletId;
+        }
+
+        var walletEndpoint = await _walletClient.CreateWalletEndpoint(receiverOrganizationId, walletId.Value, CancellationToken.None);
+
+        var externalEndpoint =
+            await _walletClient.CreateExternalEndpoint(proposal.SenderCompanyId.Value, walletEndpoint, proposal.ReceiverCompanyTin.Value,
+                CancellationToken.None);
+
+        var transferAgreement = new TransferAgreement
+        {
+            StartDate = proposal.StartDate,
+            EndDate = proposal.EndDate,
+            SenderId = proposal.SenderCompanyId,
+            SenderName = proposal.SenderCompanyName,
+            SenderTin = proposal.SenderCompanyTin,
+            ReceiverName = command.ReceiverOrganizationName,
+            ReceiverTin = command.ReceiverOrganizationTin,
+            ReceiverReference = externalEndpoint.ReceiverId
+        };
+
+        var result = await taRepo.AddTransferAgreementAndDeleteProposal(transferAgreement, command.TransferAgreementProposalId, cancellationToken);
+
+        await AppendProposalAcceptedToActivityLog(_identityDescriptor, result, proposal);
+
+        await _unitOfWork.SaveAsync();
+
+        return new AcceptTransferAgreementProposalCommandResult(result.Id, result.SenderName.Value, result.SenderTin.Value, result.ReceiverTin.Value,
+            result.StartDate.EpochSeconds, result.EndDate?.EpochSeconds, result.Type);
+    }
+
+    private async Task AppendProposalAcceptedToActivityLog(IdentityDescriptor identity, TransferAgreement result, TransferAgreementProposal proposal)
+    {
+        // Receiver entry
+        await _unitOfWork.ActivityLogEntryRepo.AddActivityLogEntryAsync(ActivityLogEntry.Create(
```

Comment:
I would delete activitylog :D Know it's not really something we can do. It's going to be so janky.